File: cpp/leetcode/254.cpp
class Solution {
public:
    vector<vector<int>> getFactors(int n) {
       vector<vector<int>> res;
       vector<int> ans;
       dfs(2, n, ans, res);
       return res;
    }
    
    void dfs(int i, int n, vector<int> &ans, vector<vector<int>> &res){
        for(;i * i <= n; i++){
            if(n % i == 0){
                ans.push_back(i); ans.push_back(n / i);
                res.push_back(ans);
                ans.pop_back();
                dfs(i, n / i, ans, res);
                ans.pop_back();
            }
        }
    }
};

File: python/leetcode/254.py
class Solution(object):
    def getFactors(self, n):
        res = []
        self.help(n, 2, [], res)
        return res

    def help(self, n, i, ans, res):
        while i*i <= n:
            if n % i == 0:
                res.append(ans + [i, n/i])
                self.help(n/i, i, ans + [i], res)
            i += 1